the-collab-lab/tcl-19-smart-shopping-list
# Style home

## Description

**This PR:** 
- Adds styling to the homepage of the application
- All views and elements have been reviewed for cross-browser compatibility including desktop and mobile views
- All views and elements have been reviewed for accessibility including:
      -semantic HTML elements used wherever possible
      -aria-labels have been included where appropriate
      -ensuring inputs and links have accessible names
      -ensuring sufficient colour contrast between background and foreground colours
- List and AddItem views have been reviewed to ensure appropriate styling
- Nav buttons on desktop view have been moved to the top of the application for better visibility, and to avoid obstructing the view of other elements of the page.

**Notes**
- The colours used ensure sufficient colour contrast, however we may amend the final colours for design purposes, maintaining the colour contrast ratio
- We are looking into using `@apply` in order to ensure the styling is as concise as possible and reduce repetitive code.


## Related Issue

Closes #35 #38 #39

## Acceptance Criteria

- Semantic HTML elements used wherever possible
- Background and foreground colours have a sufficient contrast ratio
- Buttons have an accessible name
- Form elements have associated labels
- Links have a discernible name
- The `home` Component has a styling, created with TailwindCSS
- Using [Accessibility Insights chrome extension](https://accessibilityinsights.io/) no issue occurs


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓  | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|   ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<img width="352" alt="Screenshot 2021-02-24 at 22 13 53" src="https://user-images.githubusercontent.com/24212625/109073013-b8854300-76ed-11eb-8d00-ad461105bde7.png">
<img width="338" alt="Screenshot 2021-02-24 at 22 14 03" src="https://user-images.githubusercontent.com/24212625/109073022-bcb16080-76ed-11eb-9325-2d4dba7d6d78.png">
<img width="341" alt="Screenshot 2021-02-24 at 22 14 16" src="https://user-images.githubusercontent.com/24212625/109073028-bfac5100-76ed-11eb-8ef1-0cd2f4fc3c4c.png">

### After

<img width="330" alt="Screenshot 2021-02-24 at 22 17 14" src="https://user-images.githubusercontent.com/24212625/109073315-1f0a6100-76ee-11eb-8ce1-d3274189fcfa.png">
<img width="355" alt="Screenshot 2021-02-24 at 22 16 56" src="https://user-images.githubusercontent.com/24212625/109073336-23367e80-76ee-11eb-9cac-ecdd3810073c.png">
<img width="359" alt="Screenshot 2021-02-24 at 22 16 50" src="https://user-images.githubusercontent.com/24212625/109073363-2a5d8c80-76ee-11eb-8d53-f3ed52c98b3d.png">


## Testing Steps / QA Criteria

- From your terminal, pull down this branch with git pull origin style-home and check that branch out with `git checkout style-home`
- Run the app in your browser with npm start
- View the styled homepage, create a list and add items to view the revised styled views
- Test for accessibility by running the Accessibility insights or Lighthouse extension
